tensorboard/plugins/core/core_plugin_test.py

```diff
@@ -166,111 +174,112 @@ def testIndex_returnsActualHtml(self):
     def testDataPaths_disableAllCaching(self):
         """Test the format of the /data/runs endpoint."""
         for path in ("/data/runs", "/data/logdir"):
-            response = self.logdir_based_server.get(path)
+            response = self.server.get(path)
             self.assertEqual(200, response.status_code, msg=path)
             self.assertEqual("0", response.headers.get("Expires"), msg=path)
 
-    @unittest.skipUnless(
-        ENABLE_DB_TESTS,
-        "tf.contrib is being removed and db support depends on tf.contrib. See b/147155091",
-    )
+    def testEnvironmentForWindowTitle(self):
+        """Test that the environment route correctly returns the window
+        title."""
+        parsed_object = self._get_json(self.server, "/data/environment")
+        self.assertEqual(parsed_object["window_title"], "title foo")
+
+    def testEnvironmentForLogdir(self):
+        """Test that the environment route correctly returns the logdir."""
+        parsed_object = self._get_json(self.server, "/data/environment")
+        self.assertEqual(parsed_object["data_location"], self.get_temp_dir())
+
+    def testLogdir(self):
+        """Test the format of the data/logdir endpoint."""
+        parsed_object = self._get_json(self.server, "/data/logdir")
+        self.assertEqual(parsed_object, {"logdir": self.get_temp_dir()})
+
+
+class CorePluginDbModeTest(tf.test.TestCase):
```

Comment:
Before merge, would you kindly remind me how CorePluginDbMode differs from CorePluginDbImportMode?

Reply:
It's basically the branch here:

DB import mode: https://github.com/tensorflow/tensorboard/blob/94faac5eee1aeefdba1fdd86dcd0344dca72bf2d/tensorboard/backend/application.py#L131
DB read-only mode:
https://github.com/tensorflow/tensorboard/blob/94faac5eee1aeefdba1fdd86dcd0344dca72bf2d/tensorboard/backend/application.py#L148

Both modes exercise the SQL-based read path within the plugins, which is what we actually want to test.  The difference is that DB read-only mode expects you to pass an existing SQLite DB file with the `--db` flag (and the old test was using the `create_db_writer()` code to create such a file), whereas the DB import mode takes a logdir and automatically imports the logdir event files into a temporary DB (without using `create_db_writer()` but instead implementing mostly equivalent behavior).